lib/new_relic/agent/agent_logger.rb

```diff
@@ -83,11 +83,9 @@ def backtrace_from_exception(e)
       # Allows for passing exceptions in explicitly, which format with backtrace
       def format_and_send(level, *msgs, &block)
         if block
-          if @log.send("#{level}?")
-            msgs = Array(block.call)
-          else
-            msgs = []
-          end
+          return unless @log.send("#{level}?")
+
+          msgs = Array(block.call)
```

Comment:
We mobbed as a team on this one. In the case of a block being present and the logger's level being below the threshold, we ultimately just return `nil`, so we may as well do it early here. Doing so saves us from spawning an empty array object after `#flatten` is called.

Reply:
Thank you for going over this live with us!